### PR TITLE
App crashing on multipe clicks on textView, Issue fixed

### LINKED ISSUE #1506: App crashes if user clicks twice on date textView anywhere in the app.
**Summary:** 

If the user clicks on date textView twice then the app crashes. This bug could be found anywhere in the app wherever the date textView is present.

**Steps to reproduce:** 

In any date, TextView just click twice

**Expected behaviour:** 

It shouldn't crash as it is affecting the UX.

**Observed behaviour:** 

If the user clicks on date textView twice then the app crashes. This bug could be found anywhere in the app wherever the date textView is present.

**Device and Android version:** 

Android Version 9.0
Realme 5

**Screenshots:** 

This can be found anywhere in the app wherever the date textView is present.

### PR DESCRIPTION
**Pull request by sachin-0102**
Fixes #1506 

Please Add Screenshots If there are any UI changes.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [✓] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [✓] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [✓] If you have multiple commits please combine them into one commit by squashing them.

I've added the necessary fix. What this does is that it considers only the first click and discards all the other clicks untill the user hits the back button

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewcenter/CreateNewCenterFragment.java [hunk 1]
```diff
@@ -176,14 +177,24 @@ public void inflateActivationDate() {
         tv_activationDate.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                newDatePicker.show(getActivity().getSupportFragmentManager(), FragmentConstants
-                        .DFRAG_DATE_PICKER);
+                Dialog dialogFrg=newDatePicker.getDialog();
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewcenter/CreateNewCenterFragment.java:180]: WhitespaceAround: '=' is not preceded with whitespace.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewcenter/CreateNewCenterFragment.java [hunk 2]
```diff
@@ -176,14 +177,24 @@ public void inflateActivationDate() {
...
+                if(dialogFrg!=null && dialogFrg.isShowing()) {
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewcenter/CreateNewCenterFragment.java:181]: WhitespaceAround: 'if' is not followed by whitespace.

Comment (robustTechie) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewcenter/CreateNewCenterFragment.java:181]: WhitespaceAround: '!=' is not preceded and followed with whitespace.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewcenter/CreateNewCenterFragment.java [hunk 3]
```diff
@@ -176,14 +177,24 @@ public void inflateActivationDate() {
...
+                    inputActivationDate();
+                }
             }
 
         });
 
 
     }
+    public void inputActivationDate()
+    {
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewcenter/CreateNewCenterFragment.java:193]: '{' should be on the previous line.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java [hunk 4]
```diff
@@ -289,18 +290,38 @@ private void removeExistingImage() {
 
     @OnClick(R.id.tv_submission_date)
     public void onClickTextViewSubmissionDate() {
+        Dialog dialogFrg=datePickerSubmissionDate.getDialog();
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java:293]: WhitespaceAround: '=' is not preceded with whitespace.

Comment (robustTechie) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java:293]: WhitespaceAround: '=' is not followed by whitespace.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java [hunk 5]
```diff
@@ -289,18 +290,38 @@ private void removeExistingImage() {
...
+        if(dialogFrg!=null && dialogFrg.isShowing()) {
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java:294]: WhitespaceAround: 'if' is not followed by whitespace.

Comment (robustTechie) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java:294]: WhitespaceAround: '!=' is not preceded and followed with whitespace.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java [hunk 6]
```diff
@@ -289,18 +290,38 @@ private void removeExistingImage() {
...
+            // no need to call inputSubmissionDate()
+        } else {
+            inputSubmissionDate();
+        }
+    }
+
+    public void inputSubmissionDate()
+    {
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java:302]: '{' should be on the previous line.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java [hunk 7]
```diff
@@ -289,18 +290,38 @@ private void removeExistingImage() {
...
         datePickerSubmissionDate.show(getActivity().getSupportFragmentManager(),
                 FragmentConstants.DFRAG_DATE_PICKER);
         mCurrentDateView = tvSubmissionDate;
     }
 
     @OnClick(R.id.tv_dateofbirth)
     public void onClickTextViewDateOfBirth() {
+        Dialog dialogFrg=datePickerDateOfBirth.getDialog();
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java:310]: WhitespaceAround: '=' is not preceded and followed with whitespace.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java [hunk 8]
```diff
@@ -289,18 +290,38 @@ private void removeExistingImage() {
...
+        if(dialogFrg!=null && dialogFrg.isShowing()) {
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java:311]: WhitespaceAround: 'if' is not followed by whitespace.

Comment (robustTechie) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java:311]: WhitespaceAround: '!=' is not preceded and followed with whitespace.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java [hunk 9]
```diff
@@ -289,18 +290,38 @@ private void removeExistingImage() {
...
+            // no need to call inputDateOfBirth()
+        } else {
+            inputDateOfBirth();
+        }
+    }
+    public void inputDateOfBirth()
+    {
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewclient/CreateNewClientFragment.java:318]: '{' should be on the previous line.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java [hunk 10]
```diff
@@ -218,12 +219,22 @@ public void inflateSubmissionDate() {
         tv_submissionDate.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                mfDatePicker.show(getActivity().getSupportFragmentManager(), FragmentConstants
-                        .DFRAG_DATE_PICKER);
+                Dialog dialogFrg=mfDatePicker.getDialog();
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:222]: WhitespaceAround: '=' is not preceded and followed with whitespace.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java [hunk 11]
```diff
@@ -218,12 +219,22 @@ public void inflateSubmissionDate() {
...
+                if(dialogFrg!=null && dialogFrg.isShowing()) {
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:223]: WhitespaceAround: 'if' is not followed by whitespace.

Comment (robustTechie) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:223]: WhitespaceAround: '!=' is not preceded and followed with whitespace.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java [hunk 12]
```diff
@@ -218,12 +219,22 @@ public void inflateSubmissionDate() {
...
+                // no need to call inputSubmiss ionDate()
+                } else {
+                inputSubmissionDate();
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:226]: 'else' child have incorrect indentation

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java [hunk 13]
```diff
@@ -218,12 +219,22 @@ public void inflateSubmissionDate() {
...
+                }
+
             }
         });
 
     }
+    public void inputSubmissionDate(){
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:233]: WhitespaceAround: '{' is not preceded with whitespace.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java [hunk 14]
```diff
@@ -233,15 +244,24 @@ public void inflateActivationDate() {
         tv_activationDate.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                newDatePicker.show(getActivity().getSupportFragmentManager(), FragmentConstants
-                        .DFRAG_DATE_PICKER);
+                Dialog dialogFrg=newDatePicker.getDialog();
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:247]: WhitespaceAround: '=' is not preceded and followed with whitespace.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java [hunk 15]
```diff
@@ -233,15 +244,24 @@ public void inflateActivationDate() {
...
+                if(dialogFrg!=null && dialogFrg.isShowing()) {
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:248]: WhitespaceAround: 'if' is not followed by whitespace.

Comment (robustTechie) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:248]: WhitespaceAround: '!=' is not preceded and followed with whitespace.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java [hunk 16]
```diff
@@ -233,15 +244,24 @@ public void inflateActivationDate() {
...
+                    inputActivationDate();
+                }
             }
 
         });
 
 
     }
 
+    public void inputActivationDate(){
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:260]: WhitespaceAround: '{' is not preceded with whitespace.

**Review by robustTechie — COMMENTED** [review]
I had commented a dew changes just do that the build will be successful. These problems are incurred because the build checks for a particular checkstyles.

**Comment by sachin-0102** [comment]
@robustTechie , so what you're saying is that my code is correct. It's just not alligned in the proper way to pass the build

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewcenter/CreateNewCenterFragment.java (hunk 1 above)
Comment (sachin-0102) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewcenter/CreateNewCenterFragment.java:180]: If I add whitespace before and after the'=' will it pass?

Comment (robustTechie) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewcenter/CreateNewCenterFragment.java:180]: Yeah sure

Comment (robustTechie) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewcenter/CreateNewCenterFragment.java:180]: fulfill all the comments

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java (hunk 12 above)
Comment (sachin-0102) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:226]: What does this mean?

**Comment by sachin-0102** [comment]
Hey @robustTechie ... help needed.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java [hunk 17]
```diff
@@ -218,12 +219,22 @@ public void inflateSubmissionDate() {
         tv_submissionDate.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                mfDatePicker.show(getActivity().getSupportFragmentManager(), FragmentConstants
-                        .DFRAG_DATE_PICKER);
+                Dialog dialogFrg = mfDatePicker.getDialog();
+                if (dialogFrg != null && dialogFrg.isShowing()) {
+                // no need to call inputSubmiss ionDate()
+                } else {
+                inputSubmissionDate();
```
Comment (robustTechie) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:226]: incorrect indentation of input submission date

**Review by robustTechie — CHANGES_REQUESTED** [review]

**Comment by robustTechie** [comment]
> Hey @robustTechie ... help needed.
One violation is still there

**Comment by robustTechie** [comment]
> Hey @robustTechie ... help needed.

One violation is still there

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java (hunk 17 above)
Comment (sachin-0102) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:226]: In the comment?

Comment (sachin-0102) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:226]: Or I have to add a "TAB" before the inputSubmissionDate()

Comment (robustTechie) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java:226]: tab before the function call

Comment (sachin-0102) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/online/createnewgroup/CreateNewGroupFragment.java]: okay

**Comment by sachin-0102** [comment]
Hey @robustTechie , thnaks man! This was some nastly business.

**Comment by robustTechie** [comment]
> Hey @robustTechie , thanks man! This was some nastly business.

:) Squash all the commits into one.

**Comment by sachin-0102** [comment]
> > Hey @robustTechie , thanks man! This was some nastly business.
> 
> :) Squash all the commits into one.

Done!

**Comment by sachin-0102** [comment]
Hey @robustTechie . I had some doubts. Is there something I can contact you on?

**Comment by robustTechie** [comment]
> Hey @robustTechie . I had some doubts. Is there something I can contact you on?

Yeah sure

**Comment by Aditya-gupta99** [comment]
We are closing this PR as it causes a branch conflict since the Java file has now been migrated to kotlin

**Pull request closed without merging**